### PR TITLE
Add a _render_csv method to the robot_libdoc task.

### PR DESCRIPTION
**Pull request by boakley**
This adds a feature to the robot_libdoc task. Simply by changing the output file to have a .csv suffix, a csv file
will be generated instead of the normal HTML file. 

`$ cci task run robot_libdoc --output keywords.csv`

# Critical Changes

# Changes
* A new feature has been added to the `robot_libdoc` task. If you specify the output file as a .csv file, the `robot_libdoc` task will generate a csv file with all of the keywords. 

# Issues Closed

### CODE REVIEW

File: cumulusci/tasks/robotframework/tests/test_robotframework.py [hunk 1]
```diff
@@ -458,6 +459,101 @@ def test_pageobject(self):
...
 
+    def test_csv(self):
+        path = ",".join(
+            (
+                os.path.join(self.datadir, "TestLibrary.py"),
+                os.path.join(self.datadir, "TestResource.robot"),
+                os.path.join(self.datadir, "TestPageObjects.py"),
+            )
+        )
+        output = os.path.join(self.tmpdir, "keywords.csv")
```
Comment (prescod) [inline cumulusci/tasks/robotframework/tests/test_robotframework.py:470]: I think we are moving towards Pathlib.

**Review by prescod — COMMENTED** [review]
Mostly minor suggestions.

File: docs/robot_advanced.rst [hunk 2]
```diff
@@ -339,6 +339,8 @@ If you define a robot resource file named ``MyProject.resource`` and place it in
               path: robot/MyProject/resources/MyProject.resource
               output: robot/MyProject/doc/MyProject_Library.html
 
+Normally this task will generate HTML output. If the output file ends
+with ".csv", a csv file will be generated intead.
```
Comment (davisagli) [inline docs/robot_advanced.rst:343]:
```suggestion
with ".csv", a csv file will be generated instead.
```

File: cumulusci/tasks/robotframework/libdoc.py [hunk 3]
```diff
@@ -174,5 +196,35 @@ def __init__(self, path):
...
         self.keywords = {}
 
+    @classmethod
+    def get_header(cls):
+        return ["Name", "Source", "Line#", "po type", "po_object", "Documentation"]
+
     def add_keywords(self, libdoc, page_object=tuple()):
         self.keywords[page_object] = libdoc
+
+    def to_csv(self):
```
[the remark this answers is not in the document]
Comment (boakley) [reply cumulusci/tasks/robotframework/libdoc.py]: You are correct. I renamed some of the functions to be more clear. Thanks for the suggestion!

File: cumulusci/tasks/robotframework/libdoc.py [hunk 4]
```diff
@@ -115,18 +120,35 @@ def _run_task(self):
...
 
-        return {"files": processed_files, "html": html}
+        return {"files": processed_files}
+
+    def _render_csv(self, kwfiles):
+        """Convert the list of keyword files into a list of keywords"""
+        rows = []
+        for kwfile in kwfiles:
+            rows.extend(kwfile.to_csv())
+        rows = sorted(set(map(tuple, rows)))
```
[the remark this answers is not in the document]
Comment (boakley) [reply cumulusci/tasks/robotframework/libdoc.py]: yes, good catch! I needed the map in an earlier version and then refactored to return tuples so it's no longer needed.

File: cumulusci/tasks/robotframework/tests/test_robotframework.py (hunk 1 above)
Comment (boakley) [reply cumulusci/tasks/robotframework/tests/test_robotframework.py]: Good point. I modified this new code, but I'm not going to refactor the whole test in the interest of getting this finished this sprint.

File: docs/robot_advanced.rst (hunk 2 above)
Comment (boakley) [reply docs/robot_advanced.rst]: Good catch. Thanks!

File: cumulusci/tasks/robotframework/tests/test_robotframework.py (hunk 1 above)
Comment (prescod) [reply cumulusci/tasks/robotframework/tests/test_robotframework.py]: It's a minor issue but you can replace many of your uses of os.path with pathobj.foo() but it's up to you whether you change this code today.

**Review by prescod — APPROVED** [review]

**Pull request merged**